### PR TITLE
Prepare for using "ControllerRef" in scheduler

### PR DESCRIPTION
**Pull request by wojtek-t**
This is part of a PR that I already have to avoid a bunch of rebases in the future (controller ref probably won't happen in 1.4 release).

@davidopp

### CODE REVIEW

File: plugin/pkg/scheduler/algorithm/priorities/priorities_test.go [hunk 1]
```diff
@@ -145,7 +145,11 @@ func TestZeroRequest(t *testing.T) {
 			// This should match the configuration in defaultPriorities() in
 			// plugin/pkg/scheduler/algorithmprovider/defaults/defaults.go if you want
 			// to test what's actually in production.
-			[]algorithm.PriorityConfig{{Function: LeastRequestedPriority, Weight: 1}, {Function: BalancedResourceAllocation, Weight: 1}, {Function: NewSelectorSpreadPriority(algorithm.FakePodLister(test.pods), algorithm.FakeServiceLister([]api.Service{}), algorithm.FakeControllerLister([]api.ReplicationController{}), algorithm.FakeReplicaSetLister([]extensions.ReplicaSet{})), Weight: 1}},
+			[]algorithm.PriorityConfig{
+				{Function: LeastRequestedPriority, Weight: 1},
+				{Function: BalancedResourceAllocation, Weight: 1},
+				{Function: NewSelectorSpreadPriority(algorithm.FakePodLister(test.pods), algorithm.FakeServiceLister([]api.Service{}), algorithm.FakeControllerLister([]api.ReplicationController{}), algorithm.FakeReplicaSetLister([]extensions.ReplicaSet{})), Weight: 1},
```
Comment (gmarek) [inline plugin/pkg/scheduler/algorithm/priorities/priorities_test.go]: Maybe split arguments into multiple lines as well.

File: plugin/pkg/scheduler/algorithm/priorities/selector_spreading_test.go [hunk 2]
```diff
@@ -320,8 +333,8 @@ func TestZoneSelectorSpreadPriority(t *testing.T) {
 		nodeMachine3Zone3: buildNodeLabels("zone3"),
 	}
 
-	buildPod := func(nodeName string, labels map[string]string) *api.Pod {
-		pod := &api.Pod{Spec: api.PodSpec{NodeName: nodeName}, ObjectMeta: api.ObjectMeta{Labels: labels}}
+	buildPod := func(nodeName string, labels map[string]string, ownerRefs []api.OwnerReference) *api.Pod {
```
Comment (gmarek) [inline plugin/pkg/scheduler/algorithm/priorities/selector_spreading_test.go]: Make it a free-standing function.

Comment (wojtek-t) [inline plugin/pkg/scheduler/algorithm/priorities/selector_spreading_test.go]: done

File: plugin/pkg/scheduler/algorithm/priorities/priorities_test.go (hunk 1 above)
Comment (wojtek-t) [inline plugin/pkg/scheduler/algorithm/priorities/priorities_test.go]: done

**Comment by wojtek-t** [comment]
@gmarek - PTAL

**Comment by gmarek** [comment]
LGTM

**Pull request merged**